vars/githubscm.groovy

```diff
@@ -499,3 +506,91 @@ def getLatestTag(String startsWith = '', String endsWith = '', List ignoreTags =
     return sh(returnStdout: true, script: cmd).trim()
 }
+
+/*
+* Store in env the commit info needed to update the commit status
+*/
+void prepareCommitStatusInformation(String repository, String author, String branch, String credentialsId = 'kie-ci') {
+    String tempDir = util.generateTempFolder()
+    dir(tempDir) {
+        checkout(resolveRepository(repository, author, branch, false, credentialsId))
+        setCommitStatusRepoURLEnv()
+        setCommitStatusShaEnv()
+    }
+}
+
+String getCommitStatusRepoURLEnv() {
+    return env.COMMIT_STATUS_REPO_URL
+}
+
+void setCommitStatusRepoURLEnv(String url = '') {
+    env.COMMIT_STATUS_REPO_URL = url ?: getGitRepositoryURL()
+}
+
+String getCommitStatusShaEnv() {
+    return env.COMMIT_STATUS_SHA
+}
+
+void setCommitStatusShaEnv(String sha = '') {
+    env.COMMIT_STATUS_SHA = sha ?: getCommitHash()
+}
+
+/*
+* UpdateGithubCommitStatus for the given repository
+*
+* (Run `prepareCommitStatusInformation` before if you need to set specific commit info before updating. Useful when working with `checkoutIfExists`)
+*
+*   @params checkName       Name of the check to appear into GH check status page
+*   @params state           State of the check: 'PENDING' / 'SUCCESS' / 'ERROR' / 'FAILURE'
+*   @params message         Message to display next to the check
+*/
+def updateGithubCommitStatus(String checkName, String state, String message) {
+    println "[INFO] Update commit status for check ${checkName}: state = ${state} and message = ${message}"
+    if (!getCommitStatusRepoURLEnv() || !getCommitStatusShaEnv()) {
+        println '[INFO] Commit status info are not stored, guessing from current repository'
+        setCommitStatusRepoURLEnv()
+        setCommitStatusShaEnv()
+    }
+    println "[DEBUG] repo url = ${getCommitStatusRepoURLEnv()}"
+    println "[DEBUG] commit sha = ${getCommitStatusShaEnv()}"
+
+    step([
+        $class: 'GitHubCommitStatusSetter',
+        commitShaSource: [$class: 'ManuallyEnteredShaSource', sha: env.COMMIT_STATUS_SHA],
+        contextSource: [$class: 'ManuallyEnteredCommitContextSource', context: checkName],
+        reposSource: [$class: 'ManuallyEnteredRepositorySource', url: env.COMMIT_STATUS_REPO_URL],
+        statusResultSource: [ $class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]] ],
+    ])
+}
+
+def updateGithubCommitStatusFromBuildResult(String checkName) {
+    println "[INFO] Update commit status for check ${checkName} from build result"
+    String buildResult = currentBuild.currentResult
+    println "[DEBUG] Got build result ${buildResult}"
+
+    String testsInfo = ''
+    def testResults = util.retrieveTestResults()
+    if(testResults) {
+        int pass = testResults.passCount
+        int skip = testResults.skipCount
+        int fail = testResults.failCount
+        additionalInfo = "${pass + skip + fail} tests run, ${failCount} failed, ${skipCount} skipped."
```

Comment:
shouldn't it be `testsInfo`? :thinking: something is not matching here FMPOV... :thinking: 
```suggestion
        testsInfo = "${pass + skip + fail} tests run, ${failCount} failed, ${skipCount} skipped."
```

Reply:
yeah indeed. And yesterday evening I could not see what was wrong in the tests ^^ Gonna correct this. Thanks for catching it !

Reply:
corrected